linked_list/main.py

```diff
@@ -43,53 +43,173 @@ class Node(object):
     Node class representing each of the linked nodes in the list.
     """
 
-    def __init__(self, elem, next=None):
-        pass
+    def __init__(self, elem, next=None): # jon:DONE
+        self.elem = elem
+        self.next = None
 
-    def __str__(self):
-        pass
-
-    def __eq__(self, other):
-        pass
+    def __str__(self): # jon:DONE
+        return str(self.elem)
 
-    def __repr__(self):
-        pass
+    def __eq__(self, other): # akshith: Done
+        if type(self) == type(other):
+            return self.elem == other.elem
+        else:
+            return False
 
+    def __repr__(self): # jon:DONE
+        return "Node({}, {})".format(self.elem, self.next)
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
-    def __init__(self, elements=None):
-        pass
-
-    def __str__(self):
-        pass
-
-    def __len__(self):
-        pass
-
-    def __iter__(self):
-        pass
-
-    def __getitem__(self, index):
-        pass
-
-    def __add__(self, other):
-        pass
-
-    def __iadd__(self, other):
-        pass
-
-    def __eq__(self, other):
-        pass
-
-    def append(self, elem):
-        pass
-
-    def count(self):
-        pass
-
-    def pop(self, index=None):
-        pass
+    def __init__(self, elements=None): # jon:DONE
+        #This should initialize a node for each item in the list.
+        # then populate 'elem' with the element, and link up the next node.
+
+        self.elements = elements
+        self.start = None
+        self.end = None
+
+        if self.elements:
+            # set up a new list of 1 item using first passed element.
+            self.start = Node(self.elements[0])
+            self.end = self.start
+            # trim off the first item, so we don't repeat it.
+            self.elements = self.elements[1:]
+            # now iterate and populat ethe rest of the list.
+            for item in self.elements:
+                self.append(item)
```

Comment:
Shouldn't append also work for an empty list? Couldn't you just avoid all the `self.start = Node(self.elements[0])` part?